app/scripts/modules/oracle/src/domain/IOracleLoadBalancer.ts

```diff
@@ -0,0 +1,101 @@
+import { ILoadBalancer, ILoadBalancerDeleteCommand, ILoadBalancerUpsertCommand } from '@spinnaker/core';
+import { ISubnet } from '../../../core/src/domain';
+
+export type ListenerProtocol = 'HTTP' | 'HTTPS' | 'TCP' | 'SSL';
+export enum LoadBalancingPolicy {
+  ROUND_ROBIN = 'ROUND_ROBIN',
+  IP_HASH = 'IP_HASH',
+  LEAST_CONNECTIONS = 'LEAST_CONNECTIONS',
+}
+
+export interface IOracleSubnet extends ISubnet {
+  id: string;
+  name: string;
+  availabilityDomain: string;
+  securityListIds: string[];
+  vcnId: string;
+}
+
+export interface IOracleLoadBalancer extends ILoadBalancer {
+  shape: string; // required
+  isPrivate: boolean; // required
+  subnets: IOracleSubnet[]; // required 1 for private LB, 2 for public LB
+  certificates: IOracleListenerCertificate[];
+  listeners?: { [name: string]: IOracleListener }; // not required to create LB, but useless without it (TODO Can it be added later or should we require listener?)
```

Comment:
we need to resolve the TODOs

Reply:
removed most of the TODOs.
TODOs are pathRouteSets and sessionPersistenceConfig